### PR TITLE
[NHWC support] Revoking mutually exclusive requirement on channels last and contiguous tensor

### PR DESCRIPTION
**Pull request by jjsjann123**
The old implementation assumed `is_channels_last_contiguous_` to be mutually
exclusive to `is_contiguous_`, which is not true.
Properly set the flag by checking strides.

### CODE REVIEW

**Comment by jjsjann123** [comment]
Realized that I forgot to tag @VitalyFedyunin 
Bug fixing supporting #23403

File: test/test_torch.py [hunk 1]
```diff
@@ -12520,6 +12520,14 @@ def test_memory_format(self):
         self.assertTrue(nhwc.is_contiguous(memory_format=torch.channels_last))
         self.assertEqual(nhwc, x)
 
+    def test_memory_format_consistency(self):
```
Comment (VitalyFedyunin) [inline test/test_torch.py]: Wait, does this test fails if we remove all `TensorImpl.h` changes

Comment (jjsjann123) [reply test/test_torch.py]: Yeah, that's the issue I was complaining (along with the ambiguity things) earlier last week.

I guess I didn't make it clear enough back then. :)

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by ailzhang** [comment]
cc: @VitalyFedyunin this PR is not landed yet, is this expected ? (internal diff is in preparation).

**Comment by VitalyFedyunin** [comment]
It is expected, please do not land it yet.

**Comment by VitalyFedyunin** [comment]
Please rebase and retitle to something like: 'Revoking mutually
exclusive requirement on channels last and contiguous tensor'.

**Comment by jjsjann123** [comment]
@VitalyFedyunin This one should be good to go.

**Comment by VitalyFedyunin** [comment]
Fails inner caffe2 tests, trying to figure out now what need to be fixed.

**Pull request closed without merging**